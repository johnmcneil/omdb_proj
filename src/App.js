import react, { useState } from 'react';
import './App.css';
import OmdbSearchForm from './components/OmdbSearchForm';
import MovieInfo from './components/MovieInfo';
import { Container, Jumbotron } from 'reactstrap';

function App() {
  const [title, setTitle] = useState("");

  const search = (title) => {
    setTitle(title);
  }

  return (
    <>
      <Container className="mb-3"> 
        <Jumbotron>
          <h1 className="display-3">Movie Search Tool</h1>
          <p className="lead">Search the <a href="https://www.omdbapi.com/">Open Movie Database API</a></p>
          <hr className="my-2" />
          <p>Search by title for a movie you want information about.</p>
        </Jumbotron>
      </Container>
      <OmdbSearchForm onSearch={search} />
      <MovieInfo title={title} />
    </>
  );
}

export default App;
