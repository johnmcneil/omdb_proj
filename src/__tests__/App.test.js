import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

describe('snapshot test', ()=>{
	test('testing omdb react app', () => {
		let tree = create(<App />)
		expect(tree.toJSON()).toMatchSnapshot();
	})
})