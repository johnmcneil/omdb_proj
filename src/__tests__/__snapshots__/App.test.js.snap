// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test testing omdb react app 1`] = `
Array [
  <div
    className="mb-3 container"
  >
    <div
      className="jumbotron"
    >
      <h1
        className="display-3"
      >
        Movie Search Tool
      </h1>
      <p
        className="lead"
      >
        Search the 
        <a
          href="https://www.omdbapi.com/"
        >
          Open Movie Database API
        </a>
      </p>
      <hr
        className="my-2"
      />
      <p>
        Search by title for a movie you want information about.
      </p>
    </div>
  </div>,
  <div
    className="my-3 container"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="row form-group"
      >
        <div
          className="col-4"
        >
          <input
            className="form-control"
            id="title"
            name="title"
            onChange={[Function]}
            placeholder="Movie Title"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="col-sm-1"
        >
          <button
            aria-label={null}
            className="btn btn-secondary"
            onClick={[Function]}
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>,
  <h4>
    loading...
  </h4>,
]
`;
